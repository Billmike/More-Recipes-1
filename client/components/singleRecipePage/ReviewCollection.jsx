import React from 'react';
import PropTypes from 'prop-types';

import Review from './Review';
import Spinner from '../common/Spinner';

const propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  singleRecipe: PropTypes.shape({
    isLoadingReviews: PropTypes.bool,
  }).isRequired,
};

/**
 * ReviewCollection component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} - React element
 */
const ReviewCollection = (props) => {
  const { reviews, singleRecipe } = props;
  const { isLoadingReviews } = singleRecipe;

  return (
    <div className="row">
      <div className="col s12 valign-wrapper deep-orange darken-4">
        <h5 className="white-text">Reviews</h5>
      </div>
      {reviews.length === 0 && isLoadingReviews
        ? <div className="center-align">
          <Spinner size="small" />
        </div>
        : <ul className="collection">
          {reviews && reviews.map(review => (
            <Review key={review.id} review={review} {...props} />
          ))}
        </ul>
      }
    </div>
  );
};

ReviewCollection.propTypes = propTypes;

export default ReviewCollection;
