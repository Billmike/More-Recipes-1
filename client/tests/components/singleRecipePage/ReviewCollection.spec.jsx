import React from 'react';

import ReviewCollection from '../../../components/singleRecipePage/ReviewCollection';

describe('ReviewCollection component', () => {
  it('should render component', () => {
    const props = {
      reviews: [],
      singleRecipe: {
        isLoadingReviews: false,
      }
    };
    const wrapper = shallow(<ReviewCollection {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render spinner when loading reviews', () => {
    const props = {
      reviews: [],
      singleRecipe: {
        isLoadingReviews: true,
      }
    };
    const wrapper = shallow(<ReviewCollection {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with reviews', () => {
    const props = {
      reviews: [{
        id: 1,
        content: 'A review for the recipe',
      }, {
        id: 2,
        content: 'Another review for the recipe',
      }],
      singleRecipe: {
        isLoadingReviews: false,
      },
      handleDeleteReview: jest.fn(),
    };
    const wrapper = shallow(<ReviewCollection {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Review').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});
