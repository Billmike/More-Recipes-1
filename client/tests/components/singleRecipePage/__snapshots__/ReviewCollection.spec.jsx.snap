// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCollection component should render component 1`] = `
<div
  className="row"
>
  <div
    className="col s12 valign-wrapper deep-orange darken-4"
  >
    <h5
      className="white-text"
    >
      Reviews
    </h5>
  </div>
  <div
    className="col s12 center-align"
  >
    <p>
      No reviews yet
    </p>
  </div>
  <ul
    className="collection"
  />
</div>
`;

exports[`ReviewCollection component should render component with reviews 1`] = `
<div
  className="row"
>
  <div
    className="col s12 valign-wrapper deep-orange darken-4"
  >
    <h5
      className="white-text"
    >
      Reviews
    </h5>
  </div>
  <div
    className="col s12 center-align"
  />
  <ul
    className="collection"
  >
    <Review
      handleDeleteReview={[MockFunction]}
      key="1"
      review={
        Object {
          "content": "A review for the recipe",
          "id": 1,
        }
      }
      reviews={
        Array [
          Object {
            "content": "A review for the recipe",
            "id": 1,
          },
          Object {
            "content": "Another review for the recipe",
            "id": 2,
          },
        ]
      }
      singleRecipe={
        Object {
          "isLoadingReviews": false,
        }
      }
    />
    <Review
      handleDeleteReview={[MockFunction]}
      key="2"
      review={
        Object {
          "content": "Another review for the recipe",
          "id": 2,
        }
      }
      reviews={
        Array [
          Object {
            "content": "A review for the recipe",
            "id": 1,
          },
          Object {
            "content": "Another review for the recipe",
            "id": 2,
          },
        ]
      }
      singleRecipe={
        Object {
          "isLoadingReviews": false,
        }
      }
    />
  </ul>
</div>
`;

exports[`ReviewCollection component should render spinner when loading reviews 1`] = `
<div
  className="row"
>
  <div
    className="col s12 valign-wrapper deep-orange darken-4"
  >
    <h5
      className="white-text"
    >
      Reviews
    </h5>
  </div>
  <div
    className="col s12 center-align"
  >
    <p>
      No reviews yet
    </p>
  </div>
  <div
    className="center-align"
  >
    <Spinner
      size="small"
    />
  </div>
</div>
`;
